zkevm-circuits/src/evm_circuit/execution/logs.rs

```diff
@@ -0,0 +1,614 @@
+use crate::{
+    evm_circuit::{
+        execution::ExecutionGadget,
+        param::N_BYTES_MEMORY_WORD_SIZE,
+        step::ExecutionState,
+        table::{CallContextFieldTag, RwTableTag, TxLogFieldTag},
+        util::{
+            common_gadget::SameContextGadget,
+            constraint_builder::{
+                ConstraintBuilder, StepStateTransition,
+                Transition::{Delta, To},
+            },
+            from_bytes,
+            memory_gadget::{MemoryAddressGadget, MemoryExpansionGadget},
+            sum, Cell, Word,
+        },
+        witness::{Block, Call, ExecStep, Transaction},
+    },
+    util::Expr,
+};
+use array_init::array_init;
+use eth_types::Field;
+use eth_types::{evm_types::GasCost, evm_types::OpcodeId, ToLittleEndian, ToScalar};
+use halo2_proofs::{circuit::Region, plonk::Error};
+
+#[derive(Clone, Debug)]
+pub(crate) struct LogGadget<F> {
+    same_context: SameContextGadget<F>,
+    // memory address
+    memory_address: MemoryAddressGadget<F>,
+    topics: [Cell<F>; 4],
+    topic_selectors: [Cell<F>; 4],
+
+    contract_address: Cell<F>,
+    is_static_call: Cell<F>,
+    is_persistent: Cell<F>,
+    tx_id: Cell<F>,
+    memory_expansion: MemoryExpansionGadget<F, 1, N_BYTES_MEMORY_WORD_SIZE>,
+}
+
+impl<F: Field> ExecutionGadget<F> for LogGadget<F> {
+    const NAME: &'static str = "LOG";
+
+    const EXECUTION_STATE: ExecutionState = ExecutionState::LOG;
+
+    fn configure(cb: &mut ConstraintBuilder<F>) -> Self {
+        // check opcode is in [log0,log4], already do SameContextGadget::construct
```

Comment:
remove this comment?

Reply:
removed  in [77fe7a5](https://github.com/appliedzkp/zkevm-circuits/pull/335/commits/77fe7a57618c1a13a4cfee7a439daceedc2d4cdf) thanks